src/mongocxx/private/collection.hpp

```diff
@@ -56,10 +55,10 @@ class collection::impl {
     }
 
     mongoc_collection_t* collection_t;
-    std::string database_name;
+    bsoncxx::string::view_or_value database_name;
```

Comment:
Is this really what we want? When would we want the ownership of this data to be ambiguous?